### ISSUE TITLE
Gracefully handle device out of space for sync or download

### ISSUE THREAD

**Issue by MTRichards** [owncloud/ios#466]
As an ownCloud user, I want to have my iOS device gracefully handle the case where I am out of disk space so that my device and ownCloud app do not crash when I fill storage with ownCloud synced files

Acceptance Criteria:
- ownCloud monitors amount of free disk space on the iOS device
- New setting allows user to set the amount of free space for the app to leave empty (default is 500 MB - we can adjust later if that is not enough) - the app will leave 500MB free even when reporting no more free space
- When the device recognizes that syncing or downloading any additional files requested by the user will cause an "out of space" situation, the user is warned
- Warning appears as a popup on top of the app, and says "Automatic syncing is disabled because ownCloud is out of free space. Syncing will automatically resume once enough free space is available."
- Warning has an OK button
- Once the user removes some files from the device, or changes the "free space" setting above, syncing will resume until the device again finds a free space warning.

@jancborchardt @rperezb

**Comment by rperezb** [comment]
@jancborchardt we'd need to know where to set this option, on the settings view

"New internal option allows us to set the amount of free space for the app to leave empty (default is 500 MB - we can adjust later if that is not enough) - the app will leave 500MB free even when reporting no more free space"

**Comment by rperezb** [comment]
Some doubts:
- will this option be included under the new section: sync settings
- How it has to be shown for the users? A text field with 500 MB by default? so if a user wants to disable it, they will set it to 0. Something similar to the desktop client
thx

**Comment by MTRichards** [comment]
I think it should be under sync settings, that would make the most sense, though it also has an impact for general downloads.

**Comment by MTRichards** [comment]
>How it has to be shown for the users? A text field with 500 MB by default? so if a user wants to disable it, they will set it to 0. Something similar to the desktop client thx

Ideally there is no user editable setting. I initially had it as an editable setting, but then removed the editable setting to keep things simpler/
@jancborchardt your thoughts on this?

**Comment by jancborchardt** [comment]
Yeah, I also like that we just choose 500MB as sane setting and don’t expose a setting for this. It makes no sense for people to be able to set that they want their whole storage be filled up by ownCloud – it’s just going to be frustrating.
